platform/internal/mbed_atomic_impl.h

```diff
@@ -31,6 +31,12 @@
 extern "C" {
 #endif
 
+/**
```

Comment:
This is all implementation, so shouldn't need any Doxygen marking, I think. It's implementing stuff declared in "mbed_atomic.h", which should have Doxygen bits.